From 9eaa1f070140e01930c6c0b0d6e1cd274a5c0460 Mon Sep 17 00:00:00 2001
From: Robert Schiele <rschiele@gmail.com>
Date: Fri, 13 Sep 2024 13:31:44 +0200
Subject: [PATCH] add missing includes of boost/nowide/fstream.hpp

Some files use fstream classes from boost::nowide but lack an include of
the corresponding header file.
---
 src/slic3r/GUI/Gizmos/GLGizmoSVG.cpp     | 1 +
 src/slic3r/GUI/PresetArchiveDatabase.cpp | 3 ++-
 src/slic3r/Utils/Http.cpp                | 1 +
 3 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/slic3r/GUI/Gizmos/GLGizmoSVG.cpp b/src/slic3r/GUI/Gizmos/GLGizmoSVG.cpp
index 54e601e850..51dec4021a 100644
--- a/src/slic3r/GUI/Gizmos/GLGizmoSVG.cpp
+++ b/src/slic3r/GUI/Gizmos/GLGizmoSVG.cpp
@@ -27,6 +27,7 @@
 
 #include <wx/display.h> // detection of change DPI
 #include <boost/log/trivial.hpp>
+#include <boost/nowide/fstream.hpp>
 
 #include <GL/glew.h>
 #include <chrono> // measure enumeration of fonts
diff --git a/src/slic3r/GUI/PresetArchiveDatabase.cpp b/src/slic3r/GUI/PresetArchiveDatabase.cpp
index ac94a7f6b8..4006969e75 100644
--- a/src/slic3r/GUI/PresetArchiveDatabase.cpp
+++ b/src/slic3r/GUI/PresetArchiveDatabase.cpp
@@ -16,6 +16,7 @@
 #include <boost/property_tree/json_parser.hpp>
 #include <boost/uuid/uuid.hpp>
 #include <boost/uuid/uuid_io.hpp>
+#include <boost/nowide/fstream.hpp>
 #include <cctype>
 #include <curl/curl.h>
 #include <iostream>
@@ -916,4 +917,4 @@ void PresetArchiveDatabase::sync_blocking()
 	read_server_manifest(std::move(manifest));
 }
 
-}} // Slic3r::GUI
\ No newline at end of file
+}} // Slic3r::GUI
diff --git a/src/slic3r/Utils/Http.cpp b/src/slic3r/Utils/Http.cpp
index a1c4f57c69..6895b6678b 100644
--- a/src/slic3r/Utils/Http.cpp
+++ b/src/slic3r/Utils/Http.cpp
@@ -18,6 +18,7 @@
 #include <boost/filesystem.hpp>
 #include <boost/format.hpp>
 #include <boost/log/trivial.hpp>
+#include <boost/nowide/fstream.hpp>
 
 #include <curl/curl.h>
 
